[Feat] fs_connector: adjust the fs tree to two-tier hash directory tree

hb-lee opened the pull request:
As the KV Cache files grow, the existing directory structure is prone to generating extremely large directories, which poses significant pressure on file lookup and traversal. Therefore, we adjust the single directory tree structure to a two-tier hash directory tree structure, allowing for a more even distribution of KV Cache files and avoiding the creation of large directories.
In addition, the new directory structure places KV Cache temporary files uniformly under base_path, enabling unified management of the temporary directory.

<!--  Thanks for your contribution to LMCache!  Here are some tips for you:
1. Make sure to read the Contributing Guide before submitting your PR: https://github.com/LMCache/LMCache/blob/dev/CONTRIBUTING.md
2. If this PR closes another issue, add 'Fixes #<issue number>' somewhere in the PR summary. GitHub will automatically close that issue when this PR gets merged. Alternatively, adding 'Refs #<issue number>' will not close the issue, but help provide the reviewer more context.-->

**What this PR does / why we need it**:

**Special notes for your reviewers**:

**If applicable**:

- [ ] this PR contains user facing changes - docs added
- [ ] this PR contains unit tests

maobaolong commented:
@chunxiaozheng Would you like to take a look?

chunxiaozheng commented on lmcache/v1/storage_backend/connector/fs_connector.py:
```diff
@@ -33,24 +33,21 @@ class FSConnector(RemoteConnector):
-            else [Path(base_paths_str)]
-        )
+        # group_num is a power of 2
+        self.group_num = 256
```
Could we modify this param to configurable?

hb-lee replied:
ok, thanks for reviewing. 

And as for the tmp_path, what do you think about following the code assistant's recommendations?
`
        if self.relative_tmp_dir is not None:
            tmp_path = base_path / self.relative_tmp_dir / file_name
        else:
            tmp_path = file_path.with_suffix(".tmp")
        return file_path, tmp_path
`
If we want to delete the tmp files,  we just traverse the directory tree and delete them.

maobaolong commented:
@hb-lee Thanks for this contribution, it really helps for most of FileSystem, but it would be nice if you:
1. Do not break the previous multiple base_path feature, that feature used for multi-mountpoints case.
2. To be configurable for the layout, the existing layout can be default one to keep backward-compatible, and user can define 256, 16x16, or other layout to adapt their different scenarios.

Commit:
fs_connector: support two-tier fanout tree to balance the kv cache
files

As the KV Cache files grow, the existing directory structure is prone to
generating extremely large directories, which poses significant pressure
on file lookup and traversal. Therefore, we adjust the single directory
tree structure to a two-tier hash directory tree structure, allowing for
a more even distribution of KV Cache files and avoiding the creation of
large directories.

Signed-off-by: Hongbo Li <lihb2113@outlook.com>

hb-lee committed:
Merge branch 'dev' into fstree

maobaolong commented:
Hey @hb-lee , Do you still work on this PR?
